### PR TITLE
Add openshift service mesh istio test

### PR DESCRIPTION
**Pull request by sankarpn**
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11772/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11779/
https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/11838/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpenshiftIstioMiiDomain.java [hunk 1]
```diff
@@ -0,0 +1,382 @@
...
+        "INFO", // operator pod log level
+        -1, // domainPresenceFailureRetryMaxCount
+        -1, // domainPresenceFailureRetrySeconds
+        true, // openshift istio injection
+        domainNamespace // domainNamespace
+    );
+  }
+
+  /**
+   * Create a domain using model-in-image model.
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpenshiftIstioMiiDomain.java:154]: Where we do set the 
      (a) localhostBindingsEnabled to true in domain/istio configuration.
      (b) annotate  the server pod with sidecar.istio.io/inject: "true"

File: integration-tests/pom.xml [hunk 2]
```diff
@@ -540,6 +540,15 @@
                 </includes-failsafe>
           </properties>
         </profile>
+        <profile>
```
Comment (anpanigr) [inline integration-tests/pom.xml:543]: Can we define profile with tag definition as follow ...
```
<profile>
            <id>openshift</id>
            <properties>
                <skipITs>false</skipITs>
                <groups>openshift</groups>
            </properties>
        </profile>
```

**Review by anpanigr — COMMENTED** [review]
Can we run the integration suite on kind to make sure we do not regress

File: integration-tests/pom.xml (hunk 2 above)
Comment (sankarpn) [reply integration-tests/pom.xml:543]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpenshiftIstioMiiDomain.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpenshiftIstioMiiDomain.java:161]: inside the test method after creating the domain resource

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpenshiftIstioMiiDomain.java [hunk 3]
```diff
@@ -0,0 +1,382 @@
...
+import static oracle.weblogic.kubernetes.utils.OperatorUtils.installAndVerifyOperator;
+import static oracle.weblogic.kubernetes.utils.PodUtils.getPodCreationTime;
+import static oracle.weblogic.kubernetes.utils.PodUtils.setPodAntiAffinity;
+import static oracle.weblogic.kubernetes.utils.SecretUtils.createSecretWithUsernamePassword;
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpenshiftIstioMiiDomain.java:77]: done

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpenshiftIstioMiiDomain.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItOpenshiftIstioMiiDomain.java:161]: inside test method after creating the domain resource these two parameters are set

File: integration-tests/pom.xml (hunk 2 above)
Comment (sankarpn) [reply integration-tests/pom.xml:543]: done

**Review by anpanigr — APPROVED** [review]
lgtm

**Review by maggiehe00 — APPROVED** [review]

**Pull request merged**